app/Http/Controllers/OAuthController.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/**
+ *    Copyright 2015 ppy Pty. Ltd.
+ *
+ *    This file is part of osu!web. osu!web is distributed with the hope of
+ *    attracting more community contributions to the core ecosystem of osu!.
+ *
+ *    osu!web is free software: you can redistribute it and/or modify
+ *    it under the terms of the Affero GNU General Public License version 3
+ *    as published by the Free Software Foundation.
+ *
+ *    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+ *    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *    See the GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+ */
+namespace App\Http\Controllers;
+
+use Auth;
+use Request;
+use Response;
+use Authorizer;
+use DB;
+
+class OAuthController extends Controller
+{
+    protected $section = 'oauth';
+
+    public function __construct()
+    {
+        // TODO: uh yeah, there's gotta be a better way to do this :v
```

Comment:
It's great this is mentioned in the Laravel controller documentation but not in the middleware documentation... 😌👍💯👍👌💯